tests/components/mqtt/test_vacuum.py

```diff
@@ -302,6 +302,15 @@ def mock_publish(hass):
     state = hass.states.get('vacuum.mqtttest')
     assert 'Error: Error1' == state.attributes.get(ATTR_STATUS)
 
+    message = """{
+        "error": ""
+    }"""
+    async_fire_mqtt_message(hass, 'vacuum/state', message)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
```

Comment:
why wait twice?

Reply:
@awarecan I think it was needed before #21590 was merged and mqtt state is written directly to the state machine by the MQTT platforms. It should be cleaned up across the board, but removed from this test case for now.